### PR TITLE
JBTM-3512 replace thorntail with quarkus

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3512

### CODE REVIEW

**Comment by mmusgrov** [comment]
I also wanted to improve the readme for the example. I forgot to do that so I could do that in this PR.

**Comment by mmusgrov** [comment]
@jmfinelli this PR fixes the failures we've been seeing on our CI. Please can you take a look, then hopefully we will see the job passing again.

File: rts/lra-examples/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# MicroProfile LRA Examples
+#container  MicroProfile LRA Examples
```
Comment (ochaloup) [inline rts/lra-examples/README.md:1]: Is the word `container` intentionally added here?

File: rts/lra-examples/cdi-embedded/pom.xml [hunk 2]
```diff
@@ -76,6 +67,30 @@
...
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (ochaloup) [inline rts/lra-examples/cdi-embedded/pom.xml:92]: I think the rest assured and junit5 dependencies are not needed as they are not used in the project (as far as I can understand).

File: rts/lra-examples/cdi-embedded/pom.xml [hunk 3]
```diff
@@ -89,6 +104,24 @@
...
+                    <target>8</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${surefire-plugin.version}</version>
+                <configuration>
+                    <systemPropertyVariables>
+                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        <maven.home>${maven.home}</maven.home>
```
Comment (ochaloup) [inline rts/lra-examples/cdi-embedded/pom.xml:121]: I wonder if the `maven.home` is necessary to be provided, if it's used later in the test code.

File: rts/lra-examples/cdi-participant/README.md [hunk 4]
```diff
@@ -5,44 +5,26 @@ of an API that enables loosely coupled services to coordinate long running
...
-      <executions>
-        <execution>
-          <goals>
-            <goal>package</goal>
-          </goals>
-        </execution>
-      </executions>
-    </plugin>
-```
+The cdi example is a simple JAX-RS application with one of its methods annotated with an `@LRA` annotation and other methods annotated with `@Compensate` and `@Complete`. The `@Compensate` method will be invoked by the coordinator when the LRA is closed.
```
Comment (ochaloup) [inline rts/lra-examples/cdi-participant/README.md:8]: The sentence sounds me wrong. The `@Compensate` will be invoked when LRA is canceled.

File: rts/lra-examples/cdi-participant/pom.xml [hunk 5]
```diff
@@ -101,19 +111,27 @@
             <version>${version.microprofile.lra}</version>
         </dependency>
 
+       <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
```
Comment (ochaloup) [inline rts/lra-examples/cdi-participant/pom.xml:120]: The rest-assured and junit is probably not needed.

File: rts/lra-examples/cdi-participant/src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Quarkus Configuration file
+# key = value
+
+#quarkus.arc.exclude-types=io.narayana.lra.client.internal.proxy.nonjaxrs.LRAParticipantRegistry,io.narayana.lra.filter.ServerLRAFilter,io.narayana.lra.client.internal.proxy.nonjaxrs.LRAParticipantResource
+#quarkus.arc.exclude-types=io.narayana.lra.client.internal.proxy.nonjaxrs.LRAParticipantRegistry,io.narayana.lra.filter.ServerLRAFilter,io.narayana.lra.client.internal.proxy.nonjaxrs.LRAParticipantResource
```
Comment (ochaloup) [inline rts/lra-examples/cdi-participant/src/main/resources/application.properties:5]: Should not be these two lines deleted when they're commented out?

**Review by ochaloup — CHANGES_REQUESTED** [review]
I have few minor notes which would be nice to be adjusted. Otherwise it's nice for merging by me.

File: rts/lra-examples/cdi-embedded/pom.xml (hunk 2 above)
Comment (mmusgrov) [reply rts/lra-examples/cdi-embedded/pom.xml]: +1

File: rts/lra-examples/cdi-embedded/pom.xml (hunk 3 above)
Comment (mmusgrov) [reply rts/lra-examples/cdi-embedded/pom.xml]: +1

File: rts/lra-examples/cdi-participant/README.md (hunk 4 above)
Comment (mmusgrov) [reply rts/lra-examples/cdi-participant/README.md]: +1

File: rts/lra-examples/cdi-participant/src/main/resources/application.properties (hunk 6 above)
Comment (mmusgrov) [reply rts/lra-examples/cdi-participant/src/main/resources/application.properties]: +1

**Review by ochaloup — APPROVED** [review]
When the changes do not make troubles to CI the change is fine to me.

File: rts/lra-examples/README.md (hunk 1 above)
Comment (mmusgrov) [reply rts/lra-examples/README.md]: +1

File: rts/lra-examples/cdi-participant/pom.xml (hunk 5 above)
Comment (mmusgrov) [reply rts/lra-examples/cdi-participant/pom.xml]: +1

**Comment by mmusgrov** [comment]
catelyn seems to be having issues unrelated to the PR so I will merge and then see if I can figure out whats wrong with the catelyn job.

**Pull request merged**